mne/surface.py

```diff
@@ -1697,6 +1700,211 @@ def marching_cubes(image, level):
+                        max_peak_dist=1, voxels_max=100, use_min=False):
+    """Warp a montage to a template with image volumes using SDR.
+
+    Find areas of the input volume with intensity greater than
+    a threshold surrounding local extrema near the channel location.
+    Monotonicity from the peak is enforced to prevent channels
+    bleeding into each other.
+
+    .. note:: This is likely only applicable for channels inside the brain
+              (intracranial electrodes).
+
+    Parameters
+    ----------
+    montage : mne.channels.montage.DigMontage
+        The montage object containing the channels.
+    base_image : str | pathlib.Path | nibabel.spatialimages.SpatialImage
+        Path to a volumetric scan (e.g. CT) of the subject. Can be in any
+        format readable by nibabel. Can also be a nibabel image object.
+        Local extrema (max or min) should be nearby montage channel locations.
+    %(reg_affine)s
+    %(sdr_morph)s
+    subject_from : str
+        The name of the subject used for the Freesurfer reconstruction.
+    subject_to : str
+        The name of the subject to use as a template to morph to
+        (e.g. 'fsaverage').
+    %(subjects_dir)s
+    thresh : float
+        The quantile of the image data to use to threshold the
+        channel size on the volume.
+    max_peak_dist : int
+        The number of voxels away from the channel location to
+        look in the ``image``. This will depend on the accuracy of
+        the channel locations, the default (one voxel in all directions)
+        will work only with localizations that are that accurate.
+    voxels_max : int
+        The maximum number of voxels for each channel.
+    use_min : bool
+        Whether to hypointensities in the volume as channel locations.
+        Default False uses hyperintensities.
+
+    Returns
+    -------
+    montage_warped : mne.channels.montage.DigMontage
+        The modified montage object containing the channels.
+    image_from : nibabel.spatialimages.SpatialImage
+        An image in Freesurfer surface RAS space with voxel values
+        corresponding to the index of the channel. The background
+        is 0s and this index starts at 1.
+    image_to : nibabel.spatialimages.SpatialImage
+        The warped image with voxel values corresponding to the index
+        of the channel. The background is 0s and this index starts at 1.
+    """
+    _require_version('nibabel', 'SDR morph', '2.1.0')
+    _require_version('dipy', 'SDR morph', '0.10.1')
+    from .channels import DigMontage
+    import nibabel as nib
+    from dipy.align.imwarp import DiffeomorphicMap
+
+    _validate_type(montage, DigMontage, 'montage')
+    _validate_type(base_image, nib.spatialimages.SpatialImage, 'base_image')
+    _validate_type(reg_affine, np.ndarray, 'reg_affine')
+    _check_option('reg_affine.shape', reg_affine.shape, ((4, 4),))
+    _validate_type(sdr_morph, DiffeomorphicMap, 'sdr_morph')
```

Comment:
This should really be:
```
    _validate_type(reg_affine, np.ndarray, 'reg_affine')
    _check_option('reg_affine.shape', reg_affine.shape, ((4, 4),))
    _validate_type(sdr_morph, (DiffeomorphicMap, None), 'sdr_morph')
```
because users shouldn't have to use SDR if affine is good enough. But these arguments are just passed unmodified to `apply_volume_registration` so these checks should live there (if they don't already) and here omitted (DRY):
```suggestion
```

Reply:
Good point!